homeassistant/components/deluge/__init__.py

```diff
@@ -1 +1,356 @@
-"""The deluge component."""
+"""The Deluge integration."""
+from datetime import timedelta
+import logging
+
+from deluge_client import DelugeRPCClient, FailedToReconnectException
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (
+    ATTR_TORRENT,
+    DATA_UPDATED,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    SERVICE_ADD_TORRENT,
+)
+from .errors import CannotConnect, PasswordError, UserNameError, UnknownError
+
+_LOGGER = logging.getLogger(__name__)
+
+DELUGE_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Required(CONF_HOST): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+            vol.Required(CONF_USERNAME): cv.string,
+            vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(
+                CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+            ): cv.time_period,
+        }
+    )
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.All(cv.ensure_list, [DELUGE_SCHEMA])}, extra=vol.ALLOW_EXTRA
+)
+
+PLATFORMS = ["sensor", "switch"]
+
+SERVICE_ADD_TORRENT_SCHEMA = vol.Schema(
+    {vol.Required(ATTR_TORRENT): cv.string, vol.Required(CONF_HOST): cv.string}
+)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Import the Deluge component from config."""
+    if DOMAIN in config:
+        for entry in config[DOMAIN]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
+                )
+            )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Deluge from a config entry."""
+    client = DelugeClient(hass, entry)
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = client
+
+    if not await client.async_setup():
+        return False
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    for platform in PLATFORMS:
+        await hass.config_entries.async_forward_entry_unload(entry, platform)
+
+    client = hass.data[DOMAIN].pop(entry.entry_id)
+    if client.unsub_timer:
+        client.unsub_timer()
+
+    return True
+
+
+async def get_api(hass, entry):
```

Comment:
This is not an entry

```suggestion
async def get_api(hass, data):
```

Reply:
fixed